src/cryptonote_core/blockchain.cpp

```diff
@@ -1888,37 +1900,46 @@ bool Blockchain::handle_alternative_block(const block& b, const crypto::hash& id
     // FIXME:
     // this brings up an interesting point: consider allowing to get block
     // difficulty both by height OR by hash, not just height.
+    cryptonote::alt_block_data_t alt_data            = {};
     difficulty_type main_chain_cumulative_difficulty = m_db->get_block_cumulative_difficulty(m_db->height() - 1);
     if (alt_chain.size())
     {
-      bei.cumulative_difficulty = prev_data.cumulative_difficulty;
+      alt_data.cumulative_difficulty = prev_data.cumulative_difficulty;
     }
     else
     {
       // passed-in block's previous block's cumulative difficulty, found on the main chain
-      bei.cumulative_difficulty = m_db->get_block_cumulative_difficulty(m_db->get_block_height(b.prev_id));
+      alt_data.cumulative_difficulty = m_db->get_block_cumulative_difficulty(m_db->get_block_height(b.prev_id));
     }
-    bei.cumulative_difficulty += current_diff;
+    alt_data.cumulative_difficulty += current_diff;
 
     // add block to alternate blocks storage,
     // as well as the current "alt chain" container
-    CHECK_AND_ASSERT_MES(!m_db->get_alt_block(id, NULL, NULL), false, "insertion of new alternative block returned as it already exists");
-    cryptonote::alt_block_data_t data = {};
-    data.height                       = bei.height;
-    data.cumulative_weight            = bei.block_cumulative_weight;
-    data.cumulative_difficulty        = bei.cumulative_difficulty;
-    data.already_generated_coins      = bei.already_generated_coins;
-    data.checkpointed                 = (checkpoint != nullptr);
-    m_db->add_alt_block(id, data, cryptonote::block_to_blob(bei.bl));
-    alt_chain.push_back(bei);
-    if (data.checkpointed) num_checkpoints_on_alt_chain++;
+    CHECK_AND_ASSERT_MES(!m_db->get_alt_block(id, NULL, NULL, NULL), false, "insertion of new alternative block returned as it already exists");
+    {
+      cryptonote::blobdata checkpoint_blob;
+      if (checkpoint)
+      {
+        alt_data.checkpointed = true;
+        checkpoint_blob       = t_serializable_object_to_blob(*checkpoint);
+        num_checkpoints_on_alt_chain++;
+      }
+
+      alt_data.height                   = block_height;
+      // TODO(loki): Block cumulative weight wasn't being set in the fist place in bei?
```

Comment:
Hmm, this seems like it could be problematic.  Should this be escalated to an issue to investigate?

Reply:
I've messaged some of the monero devs since I don't really have a strong understanding of block weight